UITests/BraintreeDropIn_UITests.swift

```diff
@@ -809,6 +809,163 @@ class BraintreeDropIn_ThreeDSecure_UITests: XCTestCase {
     }
 }
 
+class BraintreeDropIn_ThreeDSecure_2_UITests: XCTestCase {
+
+    var app: XCUIApplication!
+
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        app = XCUIApplication()
+        app.launchArguments.append("-EnvironmentSandbox")
+        app.launchArguments.append("-ClientToken")
+        app.launch()
+        sleep(1)
+
+        self.waitForElementToBeHittable(app.buttons["Settings"])
+        app.navigationBars.buttons["Settings"].tap()
+        if (app.switches["Use Tokenization Key"].isSelected) {
+            app.switches["Use Tokenization Key"].tap()
+        }
+
+        app.tables.cells.staticTexts["Required (3DS Transaction Option)"].tap()
+        app.tables.cells.staticTexts["Required"].tap()
+        app.navigationBars.buttons["Settings"].tap()
```

Comment:
We don't normally us UI automation to change settings when running tests, use launch arguments that are then checked in `BraintreeDemoAppDelegate.m`. When these settings are changed using the UI, the new value is written to disk and persisted between launches, so if the order of the tests change this could have unintended consequences.

We should add support for a new launch argument named something like `ThreeDSecureVersion2`. The in `BraintreeDemoAppDelegate.m` we would do

```
if ([[[NSProcessInfo processInfo] arguments] containsObject:@"-ThreeDSecureVersion2"]) {
    [[NSUserDefaults standardUserDefaults] setInteger:BraintreeDemoTransactionServiceThreeDSecureRequestedVersion2 forKey:BraintreeDemoSettingsThreeDSecureVersionDefaultsKey];
} else {
    [[NSUserDefaults standardUserDefaults] setInteger:BraintreeDemoTransactionServiceThreeDSecureRequestedVersionLegacy forKey:BraintreeDemoSettingsThreeDSecureVersionDefaultsKey];
}
```

This ensure that the value will be reset to v1 on each UI test launch so the ordering of the tests doesn't matter.

Reply:
The above snippet isn't fully correct. It should probably be

```
if ([[[NSProcessInfo processInfo] arguments] containsObject:@"-ThreeDSecureVersion2"]) {
    [[NSUserDefaults standardUserDefaults] setInteger:BraintreeDemoTransactionServiceThreeDSecureRequestedVersion2 forKey:BraintreeDemoSettingsThreeDSecureVersionDefaultsKey];
} else if ([[[NSProcessInfo processInfo] arguments] containsObject:@"-ThreeDSecureVersion1"]) {
    [[NSUserDefaults standardUserDefaults] setInteger:BraintreeDemoTransactionServiceThreeDSecureRequestedVersionLegacy forKey:BraintreeDemoSettingsThreeDSecureVersionDefaultsKey];
}
```

And we would want to change any 3DS1 UI tests to include the `-ThreeDSecureVersion1` launch argument.